python/ray/autoscaler/ray-schema.json

```diff
@@ -302,8 +302,26 @@
                        },
                        "worker_setup_commands": {
                            "$ref": "#/definitions/commands",
-                           "description": "List of common shell commands to run to setup nodes."
-                       }
+                           "description": "List of common shell commands to run to setup nodes. This overrides the global setup_commands and worker_setup_commands."
```

Comment:
isnt this nested under worker setup commands? also why does it override?

Reply:
This is just the existing behavior, with a more specific comment

Reply:
I'm not sure what you mean. The description says "it will override worker_setup_commands", but it's clearly the description for "worker_setup_commands"

Reply:
This is the node specific worker_setup_commands and it will override the global versions 

i.e.

```
setup_commands:
    - echo this won't run

worker_setup_commands:
    - echo this also won't run

available_node_types:
    custom_type:
        node_config: {}
        resources: {}
        worker_setup_commands:
           - echo only this will run

default_worker_type: custom_type
```

Reply:
@ijrsvt can you include a comment about this node specific nature as alex described in the description?

Reply:
Yep!